reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server.logging;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+ /**
+ * @author Simon Baslé
+ * @author Audrey Neveu
+ */
+public interface AccessLogFactory extends Function<AccessLogArgProvider, AccessLog> {
+
+	/**
+	 * Helper method to create an access log factory that selectively enables access logs.
+	 * <p>
+	 * Any request (represented as an {@link AccessLogArgProvider}) that doesn't match the
+	 * provided {@link Predicate} is excluded from the access log. Other requests are logged
+	 * using the default format.
+	 *
+	 * @param predicate the filter that returns {@code true} if the request should be logged, {@code false} otherwise
+	 * @return an access log factory {@link Function} to be used in {@link reactor.netty.http.server.HttpServer#accessLog(Function)}
+	 */
+	static AccessLogFactory createFilter(Predicate<AccessLogArgProvider> predicate) {
```

Comment:
Do we really need to name the methods depending on the arguments? Can't we use just `create`?

For example in `ConnectionProvider` we do not name the `create` methods depending on the arguments.
`reactor.netty.resources.ConnectionProvider#create(java.lang.String)`
`reactor.netty.resources.ConnectionProvider#create(java.lang.String, int)`

Reply:
we do not really _need_ to, it's just a practice that allows extensibility. It is one of the benefits of factory methods over constructors: you can add more factory methods later on, taking the same _type_ of parameters but with different _semantics_ that are made clearer through the factory method's name.

that said, in the current case I don't expect a high chance of adding variants later on, except maybe the `defaultFactory()` one.

Reply:
In Reactor Netty we use `create` for the most used combinations of configurations, everything should be a `Builder` if we expect to have a lot of configurations.

Reply:
ok, let's stick with the common pattern 👍

Reply:
for reference: https://github.com/reactor/reactor-netty/pull/1421#discussion_r547262245

I voted in favor of using a more specific suffix rather than just `create`.